### ISSUE TITLE
Albumart id can be negative?

### ISSUE THREAD

**Issue by tzugen** [ultrasonic/ultrasonic#434]
> If the album_id is a negative number (for example a URI like "content://media/external_primary/audio/albumart/-824043666") then the query above on line 111 actually throws an exception with a message: "java.lang.UnsupportedOperationException: Unknown or unsupported URL: content://media/external_primary/audio/albumart/-824043666"

From: https://github.com/ultrasonic/ultrasonic/pull/414#commitcomment-50040677

I would have assumed that id are always positive and an id of -1 (or -2 for that matter) denotates an invalid id.

@SaintDubious Can you find out if it is your server which returns this negative id?

**Comment by SaintDubious** [comment]
Sorry, the more I dig into this code, the more mysterious it becomes. I used curl to request the data from my server and not only are the album ids **not** negative, but they're also very sane and orderly (they start at like 200000000 and just count up). The album ids that the `contentResolver` gives me are all over the place. 

So then I started trying to figure out how the data got into the MediaStore in the first place, and I can't find anything. As far as I can tell in this code [here](https://github.com/ultrasonic/ultrasonic/blob/develop/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/MediaStoreService.kt#L64) we first write a bunch of information about the song **excluding the album id** into the MediaStore. Then we turn around and query the same exact entry we just wrote for the album id. What we get out seems like uninitialized data. Except I then tried changing it so that I do write the album id, and then when I read it out, it's not what I wrote.

So to answer your question, no, my media server (Ampache) does not return negative ids. This negative id comes from the MediaStore on this line [here](https://github.com/ultrasonic/ultrasonic/blob/develop/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/MediaStoreService.kt#L73) but I have no idea where that number comes from.
